src/pretty.rs

```diff
@@ -182,6 +209,18 @@ fn fmt(b: &mut String, limit: isize, mut width: isize, mut docs: Vector<(isize,
                         b.push_str(" ".repeat(indent as usize).as_str());
                         indent as isize
                     }
+                    Mode::Flex => {
+                        if limit - width < broken.len() as isize {
+                            remove_ws_end(b, 1);
```

Comment:
This doesn't seem right to me. Why do we have trailing whitespace? The Document algebra shouldn't know anything about the content it contains, the formatting rules should be entirely in the structure of the algebra and not conditional on whether it contains content. I think the best approach here is to not construct a Document that renders trailing whitesapce, as it currently does on master.